examples/voice-pipeline-agent/pipeline_metrics.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+from dotenv import load_dotenv
+from livekit.agents import (
+    AutoSubscribe,
+    JobContext,
+    JobProcess,
+    WorkerOptions,
+    cli,
+    llm,
+)
+from livekit.agents.pipeline import PipelineMetrics, VoicePipelineAgent
+from livekit.plugins import deepgram, openai, silero
+
+load_dotenv()
+logger = logging.getLogger("metrics-example")
+
+
+def prewarm(proc: JobProcess):
+    proc.userdata["vad"] = silero.VAD.load()
+
+
+async def entrypoint(ctx: JobContext):
+    initial_ctx = llm.ChatContext().append(
+        role="system",
+        text=(
+            "You are a voice assistant created by LiveKit. Your interface with users will be voice. "
+            "You should use short and concise responses, and avoiding usage of unpronouncable punctuation."
+        ),
+    )
+
+    await ctx.connect(auto_subscribe=AutoSubscribe.AUDIO_ONLY)
+
+    participant = await ctx.wait_for_participant()
+    agent = VoicePipelineAgent(
+        vad=ctx.proc.userdata["vad"],
+        stt=deepgram.STT(),
+        llm=openai.LLM(),
+        tts=openai.TTS(),
+        chat_ctx=initial_ctx,
+    )
+
+    @agent.on("metrics_collected")
+    def on_metrics_collected(metrics: PipelineMetrics):
+        if metrics["type"] == "vad_metrics":
+            return  # don't log VAD metrics because it is noisy
+
+        logger.info(f"metrics collected: {metrics}")
+
+        # calculate the cost of a LLM request
+        if metrics["type"] == "llm_metrics":
```

Comment:
it seems like it'd be better to aggregate these and compute total cost at the end? ideally each metric is printed out in real time (for duration/speed/etc metrics), and then the total cost is computed during context shutdown